subsys/usb/device_next/class/usbd_cdc_ecm.c

```diff
@@ -0,0 +1,724 @@
+		.bDescriptorType = USB_DESC_CS_INTERFACE,			\
+		.bDescriptorSubtype = HEADER_FUNC_DESC,				\
+		.bcdCDC = sys_cpu_to_le16(USB_SRN_1_1),				\
+	},									\
+										\
+	.if0_union = {								\
+		.bFunctionLength = sizeof(struct cdc_union_descriptor),		\
+		.bDescriptorType = USB_DESC_CS_INTERFACE,			\
+		.bDescriptorSubtype = UNION_FUNC_DESC,				\
+		.bControlInterface = 0,						\
+		.bSubordinateInterface0 = 1,					\
+	},									\
+										\
+	.if0_ecm = {								\
+		.bFunctionLength = sizeof(struct cdc_ecm_descriptor),		\
+		.bDescriptorType = USB_DESC_CS_INTERFACE,			\
+		.bDescriptorSubtype = ETHERNET_FUNC_DESC,			\
+		.iMACAddress = CDC_ECM_IMAC_BASE + n,				\
+		.bmEthernetStatistics = sys_cpu_to_le32(0),			\
+		.wMaxSegmentSize = sys_cpu_to_le16(NET_ETH_MAX_FRAME_SIZE),	\
+		.wNumberMCFilters = sys_cpu_to_le16(0),				\
+		.bNumberPowerFilters = 0,					\
+	},									\
+										\
+	.if0_int_ep = {								\
+		.bLength = sizeof(struct usb_ep_descriptor),			\
+		.bDescriptorType = USB_DESC_ENDPOINT,				\
+		.bEndpointAddress = 0x81,					\
+		.bmAttributes = USB_EP_TYPE_INTERRUPT,				\
+		.wMaxPacketSize = sys_cpu_to_le16(CDC_ECM_EP_MPS_INT),		\
+		.bInterval = CDC_ECM_EP_INTERVAL_INT,				\
+	},									\
+										\
+	.if1_0 = {								\
+		.bLength = sizeof(struct usb_if_descriptor),			\
+		.bDescriptorType = USB_DESC_INTERFACE,				\
+		.bInterfaceNumber = 1,						\
+		.bAlternateSetting = 0,						\
+		.bNumEndpoints = 0,						\
+		.bInterfaceClass = USB_BCC_CDC_DATA,				\
+		.bInterfaceSubClass = 0,					\
+		.bInterfaceProtocol = 0,					\
+		.iInterface = 0,						\
+	},									\
+										\
+	.if1_1 = {								\
+		.bLength = sizeof(struct usb_if_descriptor),			\
+		.bDescriptorType = USB_DESC_INTERFACE,				\
+		.bInterfaceNumber = 1,						\
+		.bAlternateSetting = 1,						\
+		.bNumEndpoints = 2,						\
+		.bInterfaceClass = USB_BCC_CDC_DATA,				\
+		.bInterfaceSubClass = ECM_SUBCLASS,				\
+		.bInterfaceProtocol = 0,					\
+		.iInterface = 0,						\
+	},									\
+										\
+	.if1_1_in_ep = {							\
+		.bLength = sizeof(struct usb_ep_descriptor),			\
+		.bDescriptorType = USB_DESC_ENDPOINT,				\
+		.bEndpointAddress = 0x82,					\
+		.bmAttributes = USB_EP_TYPE_BULK,				\
+		.wMaxPacketSize = sys_cpu_to_le16(CDC_ECM_EP_MPS_BULK),		\
+		.bInterval = 0,							\
+	},									\
+										\
+	.if1_1_out_ep = {							\
+		.bLength = sizeof(struct usb_ep_descriptor),			\
+		.bDescriptorType = USB_DESC_ENDPOINT,				\
+		.bEndpointAddress = 0x01,					\
+		.bmAttributes = USB_EP_TYPE_BULK,				\
+		.wMaxPacketSize = sys_cpu_to_le16(CDC_ECM_EP_MPS_BULK),		\
+		.bInterval = 0,							\
+	},									\
+										\
+	.nil_desc = {								\
+		.bLength = 0,							\
+		.bDescriptorType = 0,						\
+	},									\
+}
+
+#define USBD_CDC_ECM_DT_DEVICE_DEFINE(n)					\
+	CDC_ECM_DEFINE_DESCRIPTOR(n);						\
+	USBD_DESC_STRING_DEFINE(mac_desc_nd_##n,				\
+				DT_INST_PROP_OR(n, mac_address, "00005E005301"),\
```

Comment:
If MAC address is changed to require property this hardcoded fallback will no longer be necessary.